inc/media_offload.php

```diff
@@ -192,11 +198,49 @@ public static function instance() {
 				add_filter( 'media_row_actions', [self::$instance, 'add_inline_media_action'], 10, 2 );
 				add_filter( 'wp_calculate_image_srcset', [self::$instance, 'calculate_image_srcset'], 1, 5 );
 				add_action( 'post_updated', [self::$instance, 'update_offload_meta'], 10, 3 );
+				add_filter( 'wp_unique_filename', [self::$instance, 'unique_filename'], 10, 6 );
 			}
 		}
 		return self::$instance;
 	}
 
+	/**
+	 * Function for `wp_unique_filename` filter-hook.
+	 *
+	 * @param string        $filename                 Unique file name.
+	 * @param string        $ext                      File extension. Example: ".png".
+	 * @param string        $dir                      Directory path.
+	 * @param callable|null $unique_filename_callback Callback function that generates the unique file name.
+	 * @param string[]      $alt_filenames            Array of alternate file names that were checked for collisions.
+	 * @param int|string    $number                   The highest number that was used to make the file name unique or an empty string if unused.
+	 *
+	 * @return string
+	 */
+	public function unique_filename( $filename, $ext, $dir, $unique_filename_callback, $alt_filenames, $number ) {
+
+		if ( self::$should_not_deduplicate === true ) {
+			return $filename;
+		}
+		$sanitized_filename = str_replace( $ext, '', $filename );
+		$args = [
+			'post_type' => 'attachment',
+			'post_mime_type' => 'image',
+			'post_name__like' => $sanitized_filename,
+			'posts_per_page' => -1,
```

Comment:
Why this is required ? On a big environment this will defintely crash.

Reply:
need to get the count of images with the same name to append the number like `-1` at the end to fix the issue with images that are uploaded with the same name. The default WP deduplication is using the local files from the uploads to check but because we remove the images that no longer works. Our only way to mimic this is using the database.